### DISCUSSION TITLE
My experience with MAUI was nothing but sheer frustration

### DISCUSSION THREAD

**Discussion by iam3yal** [dotnet/maui#20212] [General] [answered]
First of all, my respect to the teams that worked on MAUI and this feedback meant only to encourage Microsoft to do a much better job and deliver an overall much better product as there is a worldwide community that trusts you! With that in mind, I offer my sincere feedback for consideration.

I really want to emphasize that I was so excited to develop my new app in .NET MAUI but then all this excitement went out the window due to two things stability and tooling.

Iirc MAUI was released with .NET 6 and after 3 releases I'd expect it would be production ready or at least stable enough but this is far from ready where with all due respect, I'm starting to think that it's just a sad joke.

I'm developing on macOS and I've tried 3 different IDEs, 2 of which they are a Microsoft product VSCode (With the .NET MAUI extension) and VS for Mac (soon to be retired) but then even on Rider it's just blows.

Sometimes the app would compile and then hang there and you don't even know that an exception was thrown or when it does work the exception doesn't even point you to the problem and worse it doesn't tell you anything useful so you could at least track down the problem! so you have to take things out to understand what piece of code is actually faulty, it's crazy.

Now I've been developing .NET apps for 20 years I have background in WPF and many other techs within the .NET ecosystem and it's the first time where I feel nothing but disappointment from the quality of the product, I haven't tried it on Windows, the experience there might be a lot better due to VS but isn't this a cross platform tech? aren't you obligated to deliver almost the same quality across platforms? 

It seems like I spend a lot of time dealing with MAUI than actually developing the app and I didn't even get started to do any complex stuff and now sadly as much as I wanted to build this app with C# I might have to reconsider, it's such a bummer.

Hoping for a better future where .NET MAUI becomes a more reliable and enjoyable platform for developers. I believe in the potential of this technology and look forward to seeing the necessary improvements that will make it a go-to choice for cross-platform development. Thank you for your dedication and efforts in advancing the .NET ecosystem.

**Comment by Xyncgas** [comment]
>Sometimes the app would compile and then hang there

On MAUI.Blazor with minimal XAML and just one webview component still happens

>or when it does work the exception doesn't even point you to the problem and worse it doesn't tell you anything useful so you could at least track down the problem

Sometimes stack trace helps, sometimes break points, sometimes are experience

>and now sadly as much as I wanted to build this app with C# I might have to reconsider

Now you can learn something completely different say creating a browser that has JS engine and 'other language' engine that can be used to script the UI

**Reply by iam3yal** [reply]
@Xyncgas 

> On MAUI.Blazor with minimal XAML and just one webview component still happens

It happens on the MAUI Native App too.

> Sometimes stack trace helps, sometimes break points, sometimes are experience

Sometimes none of that helps, unfortunately.

> Now you can learn something completely different say creating a browser that has JS engine and 'other language' engine that can be used to script the UI

I used Electron before and I'm well versed in other techs and languages but this isn't the point.

**Comment by iam3yal** [comment, marked as answer]
The good news is that I managed to make it work by removing iOS and Android from the targets and reinstalling pretty much everything.

**Reply by Xyncgas** [reply]
That's fucked up
